src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -48,137 +30,72 @@ class TaskDeploymentService {
-        albClient.createRule(CreateRuleRequest.builder().listenerArn(albListenerResponse.listeners()[0].listenerArn()).priority(getVacantPriorityValue(rulesResponse)).conditions(albRuleCondition).actions(albRuleAction).build())
+        // IAM permissions
+        parsePolicyDocuments(additionalPermissions)
+        val iamPolicy = awsCommunicator.createIamPolicy("$userName-task-role-document", taskRolePolicyString)
+        val iamRole = awsCommunicator.createIamRole("$userName-iam-role", taskAssumeRoleString)
+        awsCommunicator.attachIamPolicyToRole(iamPolicy, iamRole)
 
-        createService(ecsClusterName, userName, ecsClient, containerPort, albTargetGroupArn)
+        // ECS
+        awsCommunicator.createEcsService(ecsClusterName, "${userName}-analytical-workspace", ecsLoadBalancer)
+        val containerOverrides = buildContainerOverrides(userName, emrClusterHostName, jupyterMemory, jupyterCpu)
+        val ecsTaskRequest = awsCommunicator.buildEcsTask(ecsClusterName, "orchestration-service-analytical-workspace", iamRole.arn(), containerOverrides)
 
-        try {
-            val response = ecsClient.runTask(createRunTaskRequestWithOverrides(userName, emrClusterHostName, jupyterMemory, jupyterCpu, ecsClusterName, additionalPermissions))
-            logger.info("ECS tasks run", "instance_arns" to response.tasks().joinToString { it.containerInstanceArn() }, "task_groups" to response.tasks().joinToString { it.group() })
-        } catch (e: Exception) {
-            logger.error("Error running ECS tasks", e)
-            throw FailedToExecuteRunTaskRequestException("Error while processing the Run Task request", e)
-        }
+        awsCommunicator.runEcsTask(ecsTaskRequest)
     }
 
-    private fun createRunTaskRequestWithOverrides(userName: String, emrHostname: String, jupyterMemory: Int, jupyterCpu: Int, ecsClusterName: String, additionalPermissions: List<String>): RunTaskRequest {
+    private fun buildContainerOverrides(userName: String, emrHostname: String, jupyterMemory: Int, jupyterCpu: Int): List<ContainerOverride> {
         val usernamePair = "USER" to userName
         val hostnamePair = "EMR_HOST_NAME" to emrHostname
 
         val screenSize = 1920 to 1080
         val chromeOptsPair = "CHROME_OPTS" to arrayOf(
-            "--no-sandbox",
-            "--window-position=0,0",
-            "--force-device-scale-factor=1",
-            "--incognito",
-            "--noerrdialogs",
-            "--disable-translate",
-            "--no-first-run",
-            "--fast",
-            "--fast-start",
-            "--disable-infobars",
-            "--disable-features=TranslateUI",
-            "--disk-cache-dir=/dev/null",
-            "--test-type https://jupyterHub:8443",
-            "--kiosk",
-            "--window-size=${screenSize.toList().joinToString(",")}"
+                "--no-sandbox",
```

Comment:
I take it that this environment variable isn't part of the task definition because we are going to customise per user based on screen size? However at the moment it looks like the screen size is fixed at 1920 x 1080.  If its just the screen size that we would be changing in the future, can we not get the CHROME_OPTS environment variable from the task definition and append the --window-size option to the end to reduce the amount of data hardcoded in the app?

Reply:
I'm not sure on this point, it was in the original code before I refactored in this change. I am all for removing hardcoded things from the code, though I can't see the associated `CHROME_OPTS` in terrafrom - do these come from elsewhere?

Reply:
As per your below comment, we are not retrieving the task definition so if they are stored there we can't get to them. I am not sure that they are available in the task definition either though?

Reply:
I did it this way when I added the env variables because it was the simplest way to implement it and within the scope of the ticket. I agree it would be more elegant to source the static bit from the TD deployed by terraform and just override the screen size, however I believe it is part of another ticket

Reply:
Thanks @draagc-iw. I have raised DW-3917 to cover this change in future